spec/integration/acceptance_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'spec_helper'
+
+RSpec.describe 'Acceptance test', type: :feature do
+  let(:workdir) { 'tmp/spec' }
+  let(:noid) { '9p2909328' }
+  let(:aip_folder) { "#{workdir}/#{noid}" }
+  let(:aip_file) { "#{aip_folder}.tar" }
+  let(:log_folder) { "#{workdir}/log" }
+
+  let!(:opts) { PushmiPullyu.options.dup }
+  let!(:redis) { Redis.new }
+
+  before do
+    # setup redis and add noid to queue
+    redis.zadd 'test:pmpy_queue', 10, noid
```

Comment:
I changed yours and am doing this in the after/teardown block. So you can hopefully assume redis is always clean to begin with. Instead of being parnoid. I ran the test suite like 50 times in random orders trying to catch weird things (and i did catch a few, reason why theres a few unreleated changed in this PR)